pkg/api/context.go

```diff
@@ -86,3 +100,8 @@ func WithNamespaceDefaultIfNone(parent Context) Context {
 	}
 	return parent
 }
+
+// withTrace returns a copy of the parent in which the trace value is set
+func withTrace(parent Context, trace string) Context {
```

Comment:
Meant to be public?

... I cannot fathom why the google context package uses functions like this and not functions that take parent as a receiver.